### PR TITLE
Resource Group validation setup and teardown

### PR DESCRIPTION
**Pull request by senthilkungumaraj**
Added new script to setup and teardown resource groups for validation. Updated the pipeline yml file to use the script during validate stage.

### CODE REVIEW

File: orchestration/IntegrationService/Implementations/AzureResourceManagerDeploymentService.ps1 [hunk 1]
```diff
@@ -53,17 +53,32 @@ Class AzureResourceManagerDeploymentService: IDeploymentService {
...
+                    $resourceGroupName,
+                    $deploymentTemplate,
+                    $deploymentParameters,
+                    $location,
+                    "validate");
+        }
+        else {
+            # Fail early if the validation resource group does not
+            # exists
+            Throw "Validation resource group not setup. Create the validation resource `
```
Comment (jorgecotillo) [inline orchestration/IntegrationService/Implementations/AzureResourceManagerDeploymentService.ps1]: Add the name of the validation resource group as part of the error message.

File: archetypes/shared-services/archetypeAzureDevopsPipeline.yml [hunk 2]
```diff
@@ -8,9 +8,22 @@ variables:
...
+    pool: 
+      name: 'vdc-self-hosted'
+    steps:
+    - task: AzurePowerShell@4
+      displayName: "Setup Validation Resource Group"
+      inputs:
+        azureSubscription: 'vdc2-hub'
+        ScriptType: 'FilePath'
+        ScriptPath: 'orchestration/OrchestrationService/ValidationResourceGroupSetup.ps1'
+        ScriptArguments: '-ResourceGroupName vdc-validation-rg'
```
Comment (jorgecotillo) [inline archetypes/shared-services/archetypeAzureDevopsPipeline.yml]: what happens if a resoureGroupName is not passed?

File: orchestration/OrchestrationService/ValidationResourceGroupSetup.ps1 [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    $ResourceGroupName,
+    [Parameter(Mandatory=$false)]
+    [string]
+    $ResourceGroupLocation,
+    [Parameter(Mandatory=$false)]
+    [switch]
+    $TearDown
+)
+
+$defaultResourceGroupName = "vdc-validation-rg";
```
Comment (jorgecotillo) [inline orchestration/OrchestrationService/ValidationResourceGroupSetup.ps1]: Here you also have a default resource group name. I think this script should be agnostic on the name, or make New-Deployment agnostic of the name, the idea is to have the default name in one place only

File: orchestration/Tests/UnitTests/ValidationResourceGroupSetup.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+##
+## ValidationResourceGroup.Tests.ps1
+##
+##          The purpose of this script is to perform the unit testing for the ValidationResourceGroup Module using Pester.
+##          The script will import the ValidationResourceGroup Module and any dependency moduels to perform the tests.
+##
+########################################################################################################################
+
+
+. ../../OrchestrationService/ValidationResourceGroupSetup.ps1;
```
Comment (jorgecotillo) [inline orchestration/Tests/UnitTests/ValidationResourceGroupSetup.Tests.ps1]: Do not use fixed path, use Join-Path or [IO.Path]::Combine instead

File: orchestration/IntegrationService/Implementations/AzureResourceManagerDeploymentService.ps1 (hunk 1 above)
Comment (senthilkungumaraj) [reply orchestration/IntegrationService/Implementations/AzureResourceManagerDeploymentService.ps1]: Updated.

File: archetypes/shared-services/archetypeAzureDevopsPipeline.yml (hunk 2 above)
Comment (senthilkungumaraj) [reply archetypes/shared-services/archetypeAzureDevopsPipeline.yml]: The script is called with "null" resulting in the script creating a resource group named null.

File: orchestration/Tests/UnitTests/ValidationResourceGroupSetup.Tests.ps1 (hunk 4 above)
Comment (senthilkungumaraj) [reply orchestration/Tests/UnitTests/ValidationResourceGroupSetup.Tests.ps1]: Fixed.

File: orchestration/OrchestrationService/ValidationResourceGroupSetup.ps1 (hunk 3 above)
Comment (senthilkungumaraj) [reply orchestration/OrchestrationService/ValidationResourceGroupSetup.ps1]: Removed from New-Deployment. Instead, importing the validation resource group setup script which will expose the variable globally.

File: archetypes/shared-services/archetypeAzureDevopsPipeline.yml (hunk 2 above)
Comment (senthilkungumaraj) [reply archetypes/shared-services/archetypeAzureDevopsPipeline.yml]: This is a bug in the Azure Powershell v4 task. Even after removing the script arguments from the task, a null string is being passed as an argument to the script. Planning to add a Platform feedback for this task.

**Pull request merged**